File: node/fib.js


function fib(n) {
    let x = 1;
    let y = 1;
    const out = [];
    for (let i = 0; i < n; i++) {
        out.push(x)
        let tmp = x
        x = y
        y = y + tmp
    }
    return out
}

let n = parseInt(process.argv[2])
fib(n).forEach((x) => {
    console.log(x);
})

File: perl/fib.pl
#!/usr/bin/env perl

use strict;

sub fib($) {
  my $x = 1;
  my $y = 1;
  my @values;
  for my $i (1..$_[0]) {
    push @values, $x;
    my $tmp = $x;
    $x = $y;
    $y = $y + $tmp;
  }

  return @values;
}

foreach my $x (fib($ARGV[0])) {
  print $x,"\n";
}
